### PR TITLE
[YUNIKORN-469]Implement recycling service in PlaceholderManager that cleans up orphan placeholders

### PR DESCRIPTION
**Pull request by HuangTing-Yao**
Run a goroutine to delete orphan placeholders every 5 seconds.
Please review @yangwwei .
Thanks.

### CODE REVIEW

File: pkg/cache/placeholder_manager.go [hunk 1]
```diff
@@ -97,3 +98,22 @@ func (mgr *PlaceholderManager) setMockedClients(mockedClients *client.Clients) {
...
+func (mgr *PlaceholderManager) cleanOrphanPlaceholders() {
+	for taskID, pod := range mgr.orphanPod {
+		err := mgr.clients.KubeClient.Delete(pod)
+		if err == nil {
+			delete(mgr.orphanPod, taskID)
+		}
+	}
+}
+
+func (mgr *PlaceholderManager) Start() {
```
Comment (yangwwei) [inline pkg/cache/placeholder_manager.go:111]: we need a stop function for this too.
to stop this, you probably need a stop chan to help

File: pkg/cache/placeholder_manager.go [hunk 2]
```diff
@@ -97,3 +98,22 @@ func (mgr *PlaceholderManager) setMockedClients(mockedClients *client.Clients) {
 	defer mgr.Unlock()
 	mgr.clients = mockedClients
 }
+
+func (mgr *PlaceholderManager) cleanOrphanPlaceholders() {
```
[the remark this answers is not in the document]
Comment (HuangTing-Yao) [reply pkg/cache/placeholder_manager.go:103]: ok, done.

File: pkg/cache/placeholder_manager.go [hunk 3]
```diff
@@ -97,3 +98,22 @@ func (mgr *PlaceholderManager) setMockedClients(mockedClients *client.Clients) {
...
+	for taskID, pod := range mgr.orphanPod {
+		err := mgr.clients.KubeClient.Delete(pod)
```
[the remark this answers is not in the document]
Comment (HuangTing-Yao) [reply pkg/cache/placeholder_manager.go:110]: ok,done.

File: pkg/cache/placeholder_manager.go (hunk 1 above)
Comment (HuangTing-Yao) [reply pkg/cache/placeholder_manager.go:119]: ok,done.

File: pkg/cache/placeholder_manager_test.go [hunk 4]
```diff
@@ -173,3 +173,26 @@ func TestCleanUp(t *testing.T) {
 	assert.Equal(t, exist, false)
 	assert.Equal(t, len(placeholderMgr.orphanPod), 0)
 }
+
```
Comment (yangwwei) [inline pkg/cache/placeholder_manager_test.go:176]: could u pls add a test case to test the start/stop function?

Comment (HuangTing-Yao) [reply pkg/cache/placeholder_manager_test.go:177]: done.

**Review by yangwwei — APPROVED** [review]
+1

**Pull request merged**